splunk/src/main/java/com/splunk/PasswordCollection.java

```diff
@@ -97,11 +103,17 @@ public Password get(Object key) {
      * @return The removed credential, or null if not found.
      */
     public Password remove(String realm, String name) {
+        if(checkForWildcards()){
+            throw new IllegalArgumentException("app context must be specified when removing a password.");
+        }
         return super.remove(String.format("%s:%s:", realm, name));
     }
 
     @Override
     public Password remove(String key) {
+        if(checkForWildcards()){
```

Comment:
It might be missing context here, but why do we have to check for the wildcard when removing the password? What happens if I'm trying to remove a password that has `-` in it given that passwords with `-` have been allowed till now?

Reply:
we check for wildcards present in 'app' and 'owner' properties and not in the password itself. As we need 'app' and 'owner' values to determine from where the StoragePassword is being delete.